core/src/main/java/org/ehcache/core/internal/service/ServiceLocator.java

```diff
@@ -423,6 +425,23 @@ ServiceMap lookupDependenciesOf(ServiceMap resolved, Class<? extends Service> re
         }
       }
     }
+    final OptionalServiceDependencies optionaAnnotation = clazz.getAnnotation(OptionalServiceDependencies.class);
+    if (optionaAnnotation != null) {
+      for (String className : optionaAnnotation.value()) {
```

Comment:
Yes. Class<?>[] would be pretty here

Reply:
This is the trick - not possible... Because otherwise they would need to be in the classpath. Dependency is optional

Reply:
Ok. So the interface service is optional then. It's a bit non-conventional but I see why you need it. So ok.